### ISSUE TITLE
fix: Unable to extract uploader id

### ISSUE THREAD

**Issue by jjangga0214** [jely2002/youtube-dl-gui#484]
<img width="261" alt="스크린샷 2023-04-08 오후 9 58 01" src="https://user-images.githubusercontent.com/28584151/230722412-cca0e5bd-239e-43e8-9c20-e56617de3643.png">

Related to: https://github.com/ytdl-org/youtube-dl/issues/31530#issuecomment-1433324977

**Comment by ngergo6** [comment]
I had the same issue.

A proper fix is needed from the developer but if you want to use the app right now you can manually update yt-dlp

- download the latest binary for your platform from here: https://github.com/yt-dlp/yt-dlp/releases
- place the binary here: `<app-directory>/resources/app.asar.unpacked/binaries`
- open a command prompt or terminal in the directory and run `yt-dlp --version`
- edit the `ytdlVersion` file with the proper version number
- open the app and wait until it updates its dependencies

**Comment by dirkf** [comment]
>Related to: ytdl-org/youtube-dl#31530 (comment)

Strictly, related to https://github.com/yt-dlp/yt-dlp/issues/6247.

**Comment by StefanLobbenmeier** [comment]
I agree, it would be nice to have an option in the settings to select the release channel, so we can also support yt-dlp daily builds

**Comment by abbasally5** [comment]
If you are having this issue on linux, download the latest yt-dlp version and replacing the version in $HOME/.youtube-dl-gui/yt-dlp-unix with the newer version. You can run $HOME/.youtube-dl-gui/yt-dlp-unix --version before and after to make sure it worked

**Comment by sagarinbabel** [comment]
So I'm on macOS and it turns out there's yt-dlp_linux in the folder and not yt-dlp macos.

When I try to run yt-dlp --version it says yt-dlp isn't recognized in the terminal in the binaries folder. 

The app doesn't work anyway and the reason seems to be 

`
"errno": "ENOENT",
"code": "ENOENT",
"syscall": "spawn /Applications/Open Video Downloader.app/Contents/Resources/app.asar.unpacked/binaries/yt-dlp-unix",
"path": "/Applications/Open Video Downloader.app/Contents/Resources/app.asar.unpacked/binaries/yt-dlp-unix",`

**Comment by StefanLobbenmeier** [comment]
@sagarinbabel feel free to use my fork, that error should be fixed there

**Comment by jely2002** [comment]
Fixed in v3 now, as that properly downloads yt-dlp.
